### PR TITLE
[FEATURES] ActivateOpenWindow, OpenWindowDetection, ChildLockEnabled

### PR DESCRIPTION
**Pull request by ilueckel**
This merge requests adds following features

# Child Lock
You can now enable/disable the child lock by setting the `childLockEnabled` property.

# Open Window Detection
You can now modify the settings `tado.[Z].[X].Rooms.[Y].openWindowDetection.enabled` and `tado.[Z].[X].Rooms.[Y].openWindowDetection.timeoutInSeconds`.

Note: I did not change state attribute `write` on `enabled` in the `state_attr.js`, as the parameter is used in other places as well. I guess, you can set it via code nonetheless?

# ActivateOpenWindow
When the thermostat detects an open window, tado will send you a notification and in the app appears a button to power off the radiator/thermostat (for the period defined in `tado.[Z].[X].Rooms.[Y].openWindowDetection.timeoutInSeconds`). The new button `tado.[Z].[X].Rooms.[Y].activateOpenWindow` can trigger this app call. 

Using the button you can create a blockly automation like this:
image
This basically is the same the "Auto-Assist" does.

Note that triggering `tado.[Z].[X].Rooms.[Y].activateOpenWindow` without a detected open window by tado has no effect.

### CODE REVIEW

**Comment by HGlab01** [comment]
Hi @ilueckel ! Thanks for the PR! I just recognized the PR today and will have a look on it in the next few days!

File: main.js [hunk 1]
```diff
@@ -1056,7 +1135,7 @@ class Tado extends utils.Adapter {
 		}
 		if (method != 'get') {
 			this.apiCallinExecution = true;
-			console.log(`Body "${JSON.stringify(data)}" for API call "${url}"`);
+			this.log.debug(`Body "${JSON.stringify(data)}" for API call "${url}"`);
```
Comment (HGlab01) [inline main.js:1138]: 'console.log' was on purpose, this helps a lot for debugging sentry info. There the console log is available.
Logging both is an option.

File: main.js [hunk 2]
```diff
@@ -123,6 +128,8 @@ class Tado extends utils.Adapter {
 						const power = await this.getStateAsync(home_id + '.Rooms.' + zone_id + '.setting.power');
 						const durationInSeconds = await this.getStateAsync(home_id + '.Rooms.' + zone_id + '.overlay.termination.durationInSeconds');
 						const nextTimeBlockStart = await this.getStateAsync(home_id + '.Rooms.' + zone_id + '.nextTimeBlock.start');
+						const openWindowDetectionEnabled = await this.getStateAsync(home_id + '.Rooms.' + zone_id + '.openWindowDetection.enabled');
```
Comment (HGlab01) [inline main.js:131]: is there a reason not to put this in a separate "else if" block as  for 'childLockEnabled'?
In the last block all states are handled which trigger setZoneOverlay().
I suggest to add additional 'else if' blocks above...

File: README.md [hunk 3]
```diff
@@ -20,22 +20,26 @@ If you like my work, please feel free to provide a personal donation
...
-| tado.[x].[yyyyyy].Home.masterswitch | Turn all devices on/off |
-| tado.[x].[yyyyyy].Rooms.[z].setting.mode | AC mode (only AC devices) |
-| tado.[x].[yyyyyy].Rooms.[z].setting.fanspeed | Fanspeed (only AC devices with V3 and older versions) |
-| tado.[x].[yyyyyy].Rooms.[z].setting.fanLebel | Fanlebel (only AC devices with V3+ version) |
-| tado.[x].[yyyyyy].Rooms.[z].setting.verticalSwing | Vertical swing (only AC devices with V3+ version) |
-| tado.[x].[yyyyyy].Rooms.[z].setting.horizontalSwing | Horizontal swing (only AC devices with V3 and olderversions) |
+| State                                                                   | Description                                                                                                  |
+|-------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------|
+| tado.[x].[yyyyyy].Rooms.[z].setting.power                               | Turn device on/off                                                                                           |
```
Comment (HGlab01) [inline README.md:25]: Which advantage do you see in adding the whites paces?
On my laptop screen one row is now displayed in 2 lines, which makes it harder to read

File: main.js (hunk 1 above)
Comment (ilueckel) [reply main.js:1138]: Makes sense. I used the iobroker logs to debug everything, therefore I changed it

File: main.js (hunk 2 above)
Comment (ilueckel) [reply main.js:131]: Okay, I wasn't sure about that. I added two new if-blocks for `activateOpenWindow` and `openWindowDetection`-settings

File: README.md (hunk 3 above)
Comment (ilueckel) [reply README.md:25]: Wasn't done on purpose, IntelliJ seems to update this automatically. As long as you don't wrap lines in the IDE, it increases readability as all columns are aligned. I changed it back

**Comment by ilueckel** [comment]
Hi @HGlab01,
I was wondering if there is anything left or needs to be changed in handling the states from your point of view?

**Comment by HGlab01** [comment]
Really nice implementations!
I hope it was not to hard to understand the general approaches for your additional implementations!

@state enabled:
I did the following "trick" for the attribute celsius:
https://github.com/DrozmotiX/ioBroker.tado/blob/dde43947766ba063d7905322ef699ae836e74e6e/main.js#L821-L824
With this I renamed the JSON attribute before creating states. I think you can do this here as well.
Maybe not extremely beautiful but it works ;-)

**Comment by ilueckel** [comment]
Great - yeah this would be a good possibility. I renamed `openWindowDetection.enabled` to `openWindowDetection.openWindowDetectionEnabled` and added the matching state.

**Comment by HGlab01** [comment]
Thanks for all the work! Let's merge to main and create an alpha version including your PR!

**Comment by HGlab01** [comment]
One additional remark/wish:
If you add additional things please deactivate sentry in your test installation. Otherwise I have a lot of issues in Sentry which makes me nervous until I recognize where it comes from ;-)

**Comment by HGlab01** [comment]
Enjoy!
https://github.com/DrozmotiX/ioBroker.tado/archive/refs/tags/v0.3.15-alpha.2.tar.gz

**Pull request merged**